src/builtin.cpp

```diff
@@ -2172,71 +2172,31 @@ enum status_cmd_t {
     STATUS_SET_JOB_CONTROL,
     STATUS_PRINT_STACK_TRACE
 };
-
-static status_cmd_t status_string_to_cmd(const wchar_t *status_command) {
-    if (wcscmp(status_command, L"is-login") == 0) {
-        return STATUS_IS_LOGIN;
-    } else if (wcscmp(status_command, L"is-interactive") == 0) {
-        return STATUS_IS_INTERACTIVE;
-    } else if (wcscmp(status_command, L"is-block") == 0) {
-        return STATUS_IS_BLOCK;
-    } else if (wcscmp(status_command, L"is-command-sub") == 0) {
-        return STATUS_IS_COMMAND_SUB;
-    } else if (wcscmp(status_command, L"is-full-job-control") == 0) {
-        return STATUS_IS_FULL_JOB_CTRL;
-    } else if (wcscmp(status_command, L"is-interactive-job-control") == 0) {
-        return STATUS_IS_INTERACTIVE_JOB_CTRL;
-    } else if (wcscmp(status_command, L"is-no-job-control") == 0) {
-        return STATUS_IS_NO_JOB_CTRL;
-    } else if (wcscmp(status_command, L"current-filename") == 0) {
-        return STATUS_CURRENT_FILENAME;
-    } else if (wcscmp(status_command, L"current-line-number") == 0) {
-        return STATUS_CURRENT_LINE_NUMBER;
-    } else if (wcscmp(status_command, L"job-control") == 0) {
-        return STATUS_SET_JOB_CONTROL;
-    } else if (wcscmp(status_command, L"print-stack-trace") == 0) {
-        return STATUS_PRINT_STACK_TRACE;
-    }
-    return STATUS_NOOP;
-}
-
-static const wcstring status_cmd_to_string(status_cmd_t status_cmd) {
-    switch (status_cmd) {
-        case STATUS_NOOP:
-            return L"no-op";
-        case STATUS_IS_LOGIN:
-            return L"is-login";
-        case STATUS_IS_INTERACTIVE:
-            return L"is-interactive";
-        case STATUS_IS_BLOCK:
-            return L"is-block";
-        case STATUS_IS_COMMAND_SUB:
-            return L"is-command-sub";
-        case STATUS_IS_FULL_JOB_CTRL:
-            return L"is-full-job-control";
-        case STATUS_IS_INTERACTIVE_JOB_CTRL:
-            return L"is-interactive-job-control";
-        case STATUS_IS_NO_JOB_CTRL:
-            return L"is-no-job-control";
-        case STATUS_CURRENT_FILENAME:
-            return L"current-filename";
-        case STATUS_CURRENT_LINE_NUMBER:
-            return L"current-line-number";
-        case STATUS_SET_JOB_CONTROL:
-            return L"job-control";
-        case STATUS_PRINT_STACK_TRACE:
-            return L"print-stack-trace";
-    }
-}
+enum_map<status_cmd_t> status_enum_map[] = {
+    {STATUS_NOOP, L"no-op"},
```

Comment:
Correct. I debated whether calling it DEFAULT or UNDEF (or any of several other variants). It's the first entry because I really dislike using zero as the first legitimate enum value (I've seen that cause too many problems). But in hindsight I'm going to rename it and make it the sentinel enum and use the `enum_name = 1,` syntax for the first legitimate value.